mani_skill/envs/tasks/dexterity/insert_flower.py

```diff
@@ -0,0 +1,206 @@
+from pathlib import Path
+from typing import Any, Dict, List, Union
+
+import numpy as np
+import sapien
+import sapien.physx as physx
+import torch
+import torch.nn.functional as F
+
+import mani_skill.envs.utils.randomization as randomization
+from mani_skill import ASSET_DIR
+from mani_skill.agents.robots import FloatingInspireHandRight
+from mani_skill.envs.sapien_env import BaseEnv
+from mani_skill.sensors.camera import CameraConfig
+from mani_skill.utils import common, sapien_utils
+from mani_skill.utils.building import actors
+from mani_skill.utils.building.actors import build_cube
+from mani_skill.utils.geometry.rotation_conversions import quaternion_apply
+from mani_skill.utils.io_utils import load_json
+from mani_skill.utils.registration import register_env
+from mani_skill.utils.scene_builder.table import TableSceneBuilder
+from mani_skill.utils.structs.actor import Actor
+from mani_skill.utils.structs.pose import Pose, vectorize_pose
+from mani_skill.utils.structs.types import Array, GPUMemoryConfig, SimConfig
+
+
+@register_env(
+    "InsertFlower-v1",
+    max_episode_steps=300,
```

Comment:
I think it would be good to include the oakink-v2 assets. We usually define new assets here: https://github.com/haosulab/ManiSkill/blob/main/mani_skill/utils/assets/data.py

Can you add a new data group for that? You can follow how e.g. the scene datasets are defined which are hosted on hugging face. We expect a zip file with all the data we need to extract into a folder.

Finally after doing so to ensure users don't run this env without downloading assets first, can you add a `asset_download_ids=[asset_id])` to the register_env function? 

https://github.com/haosulab/ManiSkill/blob/main/mani_skill/envs/tasks/tabletop/pick_single_ycb.py#L26C57-L26C84 is a example of how it is done.